pkg/image/registryclient/alternativesources.go

```diff
@@ -18,56 +18,42 @@ import (
 
 // AlternativeImageSources holds ImageContentSourcePolicy variables to look up image sources
 type AlternativeImageSources struct {
-	ICSPFile          string
-	ICSPList          []operatorv1alpha1.ImageContentSourcePolicy
-	LookupClusterICSP bool
-	ICSPClientFn      func() (operatorv1alpha1client.ImageContentSourcePolicyInterface, error)
+	ICSPFile     string
+	ICSPList     []operatorv1alpha1.ImageContentSourcePolicy
+	ICSPClientFn func() (operatorv1alpha1client.ImageContentSourcePolicyInterface, error)
 }
 
 // WithICSP returns a Context with ImageContentSourcePolicy variables populated
-func (c *Context) WithICSP(icspFile string, lookupClusterICSP bool, icspClientFn func() (operatorv1alpha1client.ImageContentSourcePolicyInterface, error)) *Context {
+func (c *Context) WithICSP(icspFile string, icspClientFn func() (operatorv1alpha1client.ImageContentSourcePolicyInterface, error)) *Context {
 	c.ImageSources = AlternativeImageSources{
-		ICSPFile:          icspFile,
-		LookupClusterICSP: lookupClusterICSP,
-		ICSPClientFn:      icspClientFn,
+		ICSPFile:     icspFile,
+		ICSPClientFn: icspClientFn,
 	}
 	return c
 }
 
-// AddICSPsFromCluster will lookup ICSPs from cluster. Unless explicit lookup is set to false, logs errors rather than returning errors.
-func (a *Context) AddICSPsFromCluster(explicit bool) error {
-	icspClient, err := a.ImageSources.ICSPClientFn()
+// addICSPsFromCluster will lookup ICSPs from cluster. Logs errors rather than returning errors. By design,
+// this function has no way of knowing whether it expects to connect to a cluster or not. That is up the the caller.
+func (a *Context) addICSPsFromCluster(icspClient operatorv1alpha1client.ImageContentSourcePolicyInterface) {
+	if icspClient == nil {
+		klog.V(4).Infof("no client exists to access ImageContentSourcePolicies in cluster")
+		return
+	}
+	icsps, err := icspClient.List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
-		return err
+		// may or may not have access to ICSPs in cluster
+		// don't error if can't access ICSPs
+		klog.V(4).Infof("did not access any ImageContentSourcePolicies in cluster: %v", err)
```

Comment:
Return early is the rule, so I'd just add `return` here, log is fine.

Reply:
fixed, updated.